charmdir.go

```diff
@@ -16,26 +16,38 @@ import (
 	"syscall"
 )
 
-// defaultIgnoreDirectories defines all directories (currently common vcs)
-// that should be ignored when archiving
-var defaultIgnoreDirectories = []string{
-	".git",
-	".svn",
-	".hg",
-	".bzr",
-}
+// defaultJujuIgnore contains jujuignore directives for excluding VCS- and
+// build-related directories when archiving. The following set of directives
+// will be appended to the contents of the charm's .jujuignore file if one is
+// provided.
+//
+// NOTE: writeArchive auto-generates its own revision and version files so they
+// need to be excluded here to prevent anyone from overriding their contents by
+// adding files with the same name to their charm repo.
+var defaultJujuIgnore = `
```

Comment:
From the bug report, it looks like tox is being noisy. Should we add `.tox` (and maybe `.tmp` as [it looks like that's used also](https://tox.readthedocs.io/en/latest/config.html#tox-global-settings))?

Reply:
The idea is that we will be updating https://github.com/juju-solutions/template-reactive-python to include a `.jujuignore` file that excludes `.tox` so that should take care of that.

Reply:
I could add all these bits to the default list but I am concerned it would probably defeat the purpose of having a jujuignore file in the first place.